connection_manager.go

```diff
@@ -308,6 +308,15 @@ func (cm *connectionManager) expireConnections() {
 
 			logDebug("[connectionManager]", "(%v) expired %d connections.", cm, expiredCount)
 
+			for cm.connectionCount < cm.minConnections {
```

Comment:
This could probably be refactored out into a function that either `start()` uses, or possibly rename this function to something along the lines of `manageConnectionPool()` which would be intended to manage all connection pool lifecycle events.